lib/optimizely/event_builder.rb

```diff
@@ -6,28 +6,18 @@ module Optimizely
   class Event
     # Representation of an event which can be sent to the Optimizely logging endpoint.
 
-    # Event API format
-    OFFLINE_API_PATH = 'https://%{project_id}.log.optimizely.com/event'
-
-    # Gets/Sets event params.
-    attr_accessor :params
+    attr_reader :method
+    attr_reader :params
+    attr_reader :url
 
-    def initialize(params)
+    def initialize(method, url, params)
+      @method = method
+      @url = url
       @params = params
     end
-
-    def url
-      # URL for sending impression/conversion event.
-      #
-      # project_id - ID for the project.
-      #
-      # Returns URL for event API.
-
-      sprintf(OFFLINE_API_PATH, project_id: @params[Params::PROJECT_ID])
-    end
   end
 
-  class EventBuilder
+  class EventBuilderV1
```

Comment:
I disagree because when we rip out V1 support, we would just be changing the internals of this class and the API references. If we keep two classes around there will also be some forking logic throughout the codebase that we need to kill. Providing the event builder as  some sort of facade/factory will make it much easier to deal with.